R/events.R

```diff
@@ -1,12 +1,16 @@
 create_events <- function(parameters) {
   events <- list(
+    # Disease progression events
```

Comment:
We re-order events to deal with the case where individuals are scheduled to progress to the next stage of disease AND be re-infected.

asymptomatic_progression (D -> A), subpatent_progression (A -> U) and recovery (U -> S) are now earlier in the event list than clinical_infection (S, U, A) -> D and asymptomatic_infection (S, U -> A).

So If an individual is due to progress from U -> S AND be infected to D at the same timestep. The clinical_infection updates will overwrite the subpatent_progression updates.

Reply:
I'm wondering if this still does not capture the competing hazards completely?
If I've understood correctly this is saying that the hazard of infection dominates the "hazard" of  recovery (U->S)?
I don't know how it was dealt with previously, but I was thinking of the following:
```
# Probability of infection
prob_infection = 0.25
# Probability of recovery
prob_recover = 0.5
# Probability of infection OR recovery 
prob_either = 1 - (1 - prob_infection) * (1 - prob_recover)
# Draw if either happens to the individual
runif(1) < prob_either
# If either does happen, sample which of the two outcomes occurs:
sample.int(2, c(prob_infection, prob_recover))
```